File: benchmarks/javascript/jobs/json_parse.js
function json_parse() {
  const start = performance.now();

  const data = [];

  for (let i = 0; i < 500000; i++) {
    data.push({
      id: i,
      arr: Array.from({ length: 20 }, (_, x) => x),
    });
  }

  const raw = JSON.stringify(data);
  const parsed = JSON.parse(raw);

  let sum = 0;

  for (let i = 0; i < parsed.length; i++) {
    sum += parsed[i].arr.length;
  }

  const end = performance.now();

  return {
    success: true,
    timeElapsedMs: end - start,
    memDeltaKb: 0,
  };
}

module.exports = { json_parse };

File: benchmarks/rust/src/jobs/json_parse.rs
use serde_json::json;
use std::time::Instant;

pub fn run() -> serde_json::Value {
    let start = Instant::now();

    let mut data = Vec::with_capacity(500_000);

    for i in 0..500_000 {
        let arr: Vec<i32> = (0..20).collect();
        data.push(serde_json::json!({
            "id": i,
            "arr": arr
        }));
    }

    let raw = serde_json::to_string(&data).unwrap();
    let parsed: serde_json::Value = serde_json::from_str(&raw).unwrap();

    let mut _sum = 0;

    for item in parsed.as_array().unwrap() {
        _sum += item["arr"].as_array().unwrap().len();
    }

    let elapsed = start.elapsed().as_millis();

    json!({
        "success": true,
        "timeElapsedMs": elapsed,
        "memDeltaKb": 0
    })
}
